azkaban-common/src/main/java/azkaban/executor/DisabledJob.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Disabled job, or disabled sub-jobs for an embedded flow.
+ */
+public class DisabledJob {
+
+  static final private String SUBFLOW_ID_KEY = "id";
+  static final private String SUBFLOW_CHILDREN_KEY = "children";
+
+  final private String name; // name of the disabled job, or embedded flow
+  final private List<DisabledJob> children; // disabled sub-jobs for an embedded flow
+
+  /**
+   * Constructor.
+   *
+   * @param name the name of the disabled job or embedded flow.
+   * @param children if an embedded flow, the disabled jobs for the embedded flow. null if it is
+   * a job.
+   */
+  public DisabledJob(String name, List<DisabledJob> children) {
+    this.name = name;
+    this.children = children;
+  }
+
+
+  /**
+   * Constructor.
+   *
+   * @param name the name of the disabled job.
+   */
+  public DisabledJob(String name) {
+    this(name, null);
+  }
+
+  public String getName() {
+    return this.name;
+  }
+
+  public List<DisabledJob> getChildren() {
+    return this.children;
+  }
+
+  /** @return True if this is a job, false if it is an embedded flow. */
+  public boolean isJob() {
```

Comment:
consider isEmbeddedFlow to be more precise?

Reply:
Will replace.